app/src/main/java/org/transdroid/search/Nebulance/NebulanceAdapter.java

```diff
@@ -0,0 +1,110 @@
+package org.transdroid.search.Nebulance;
+
+import android.annotation.SuppressLint;
+import android.content.SharedPreferences;
+
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.transdroid.search.AbstractHtmlAdapter;
+import org.transdroid.search.SearchResult;
+import org.transdroid.search.SortOrder;
+import org.transdroid.search.TorrentSite;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Adapter for Nebulance Private site.
+ */
+public class NebulanceAdapter extends AbstractHtmlAdapter {
+    private static final String BASE_URL = "https://nebulance.io/";
+    private static final String LOGIN_URL = BASE_URL + "login.php";
+    private static final String QUERY_URL = BASE_URL + "torrents.php?order_by=time&order_way=desc&searchtext=%s&search_type=0&taglist=%s&tags_type=0";
+    private static final Pattern QUALITY_REGEX = Pattern.compile("\\b(\\d+p)\\b");
+    @SuppressLint("SimpleDateFormat")
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("MMM dd yyyy, HH:mm");
+
+    @Override
+    protected String getLoginUrl() {
+        return LOGIN_URL;
+    }
+
+    @Override
+    protected String getSearchUrl(SharedPreferences prefs, String query, SortOrder order, int maxResults) throws UnsupportedEncodingException {
+        final Matcher matcher = QUALITY_REGEX.matcher(query);
```

Comment:
So this converts keywords added in the query such as (1080p) into tags for the search query?

Reply:
Exactly. This tracker unfortunately doesn't include the quality in the item name, just as a tag. So searching for a specific quality doesn't work unless you do this.

Quality seems to me to be an important thing to search by so I added this.